examples/calibrate_tsqrtP.py

```diff
@@ -0,0 +1,328 @@
+r"""
+Calculate actual beta as a function of pulse length
+===================================================
+Assuming the data is a series of pulse lengths captured on the GDS oscilloscope
+(acquired using `FLInst/examples/run_pulse_calibration.py`), here each pulse
+length capture is frequency filtered and the absolute is integrated and
+converted to yield :math:`t_{pulse}\sqrt{P_{peak}}`. To convert to rms power
+this value is divided by :math:`\sqrt{2}` to give the final beta where
+:math:`\beta = \frac{1}{\sqrt{2}} \int \sqrt{P_{rms}(t)} dt`. From these
+integrals the programmed pulse length is plotted as a function of the
+calculated :math:`\beta`. The fit is calculated and the fitting coefficients
+are output for the nonlinear and linear regime of the data. The user then
+copies these coefficients into FLInst/spincore_pp/pulse_length_conv.py in the
+appropriate section to obtain calibrated pulse lengths for the given amplitude.
+"""
+
+import pyspecdata as psd
+import matplotlib.pyplot as plt
+from pyspecProcScripts import find_apparent_anal_freq
+import numpy as np
+import logging
+
+psd.init_logging()
+V_atten_ratio = 102.2  # attenutation ratio
+skip_plots = (
+    45  # diagnostic -- set this to None, and there will be no diagnostic plots
+)
+HH_slicewidth = 1e6
+typical_180 = 40e-6  # typical beta for a 180 -- it's really important to get pulses in this regime correct
+
+# Note: the linear threshold seems to vary from one amplitude to the next
+with psd.figlist_var() as fl:
+    for filename, nodename, linear_threshold in [
+        (
+            "240819_test_amp0p05_calib_pulse_calib.h5",
+            "pulse_calib_3",
+            150e-6,
+        ),
+        (
+            "240819_amp0p1_calib_pulse_calib.h5",
+            "pulse_calib_1",
+            270e-6,
+        ),
+        (
+            "240819_amp0p2_calib_repeat_pulse_calib.h5",
+            "pulse_calib_8",
+            310e-6,
+        ),
+    ]:
+        d = psd.find_file(
+            filename, expno=nodename, exp_type="ODNP_NMR_comp/test_equipment"
+        )
+        assert (
+            d.get_prop("postproc_type") == "GDS_capture_v1"
+        ), "The wrong postproc_type was set so you most likely used the wrong script for acquisition"
+        amplitude = d.get_prop("acq_params")["amplitude"]
+        fl.basename = f"amplitude = {amplitude:.2f}"
+        # {{{ ensure units are set
+        if d.get_units("t") is None:
+            logging.info(
+                "Units for your t axis weren't set to anything so I am setting them to s"
+            )
+            d.set_units("t", "s")
+        if d.get_units("t_pulse") is None:
+            logging.info(
+                "Units for your t_pulse axis weren't set to anything so I am setting them to s"
+            )
+            d.set_units("t_pulse", "s")
```

Comment:
"if the units are not set, something is wrong --> issue a warning
only set the units if this is None --> if it's set to something else, don't mess with it"